### PR TITLE
Attempt to modernize sbt

### PR DESCRIPTION
**Pull request by tmccombs**
I'm afraid this may require us to upgrade the play version of piezo...

### CODE REVIEW

File: project/build.sbt [hunk 1]
```diff
@@ -1,13 +1,11 @@
-addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.1")
+addSbtPlugin("com.jsuereth" % "sbt-pgp" % "2.0.1")
 
-addSbtPlugin("com.lucidchart" % "sbt-cross" % "3.0")
+addSbtPlugin("com.lucidchart" % "sbt-cross" % "4.0")
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.3.9")
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.1.6")
```
Comment (tmccombs) [inline project/build.sbt:7]: I have no idea why we had this.

File: project/build.sbt [hunk 2]
```diff
@@ -1,13 +1,11 @@
-addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.1")
+addSbtPlugin("com.jsuereth" % "sbt-pgp" % "2.0.1")
 
-addSbtPlugin("com.lucidchart" % "sbt-cross" % "3.0")
+addSbtPlugin("com.lucidchart" % "sbt-cross" % "4.0")
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.3.9")
```
Comment (tmccombs) [inline project/build.sbt:5]: I don't think this version of the play plugin is compatible with sbt 1.0+. But upgrading to a more recent version of play is probably going to be more work :(

File: project/build.sbt (hunk 1 above)
Comment (sriraamas) [reply project/build.sbt:7]: Looks like we must have added it, because, our current install process uses deb packages published to bintray

Comment (tmccombs) [reply project/build.sbt:7]: oh, maybe we just need to update this to a newer version that supports sbt 1.0

File: build.sbt [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-  serverLoading in Debian := ServerLoader.Systemd,
-  target := baseDirectory.value / "target-xenial",
-  scalaVersion := "2.11.8"
-)
-lazy val `admin-bionic` = admin.copy(id = "admin-bionic").settings(
-  serverLoading in Debian := ServerLoader.Systemd,
-  target := baseDirectory.value / "target-bionic",
-  scalaVersion := "2.11.8"
-)
+// lazy val `admin-xenial` = admin.copy(id = "admin-xenial").settings(
```
Comment (sriraamas) [inline build.sbt:12]: I was meaning to dig into fixing these, after getting piezo-admin building successfully.

**Comment by rjwirth** [comment]
I'd like to mention here that this branch worked out of the box, ran both worker and admin, then switched the quartz props to my mysql instance and quartz name, and my jobs showed up in the UI.
THANK YOU!

File: project/build.sbt (hunk 1 above)
Comment (tmccombs) [reply project/build.sbt:7]: we're going to build debs differently now, so I ended up removing this.

File: project/build.sbt (hunk 2 above)
Comment (tmccombs) [reply project/build.sbt:5]: We've upgraded to play 2.8

File: build.sbt (hunk 3 above)
Comment (tmccombs) [reply build.sbt]: we're going to do debs differently now. (I'll do that in a different PR).

File: worker/src/main/scala/com/lucidchart/piezo/Worker.scala [hunk 4]
```diff
@@ -58,7 +58,6 @@ object Worker {
       scheduler.start()
       logger.info("scheduler started")
       val reader = new InputStreamReader(System.in)
-      val stdin = new BufferedReader(reader)
```
Comment (sriraamas) [inline worker/src/main/scala/com/lucidchart/piezo/Worker.scala:61]: am guessing unused?

File: build.sbt [hunk 5]
```diff
@@ -24,39 +22,8 @@ inThisBuild(Seq(
...
-      "gpgSign" -> true,
-      "vcs_tag" -> version.value
-    )
-  )
-  val file = target.value / Debian.name / "bintray.json"
-  file.getParentFile.mkdirs()
-  IO.write(file, Json.prettyPrint(json))
-  file
-}
+publishTo := sonatypePublishToBundle.value
```
Comment (sriraamas) [inline build.sbt:29]: do we still need this?

Comment (sriraamas) [reply build.sbt:29]: might be better to some one reuse commonSettings

File: worker/src/main/scala/com/lucidchart/piezo/Worker.scala (hunk 4 above)
Comment (tmccombs) [reply worker/src/main/scala/com/lucidchart/piezo/Worker.scala:61]: yep

File: build.sbt (hunk 5 above)
Comment (tmccombs) [reply build.sbt:29]: I think so, but I'm not 100% sure.

**Review by sriraamas — APPROVED** [review]

**Pull request merged**